toolchain/check/handle_name.cpp

```diff
@@ -222,17 +232,30 @@ auto HandlePointerMemberAccessExpr(Context& context, Parse::NodeId parse_node)
   return context.TODO(parse_node, "HandlePointerMemberAccessExpr");
 }
 
+static auto GetParseNodeAsName(Context& context, Parse::NodeId parse_node)
```

Comment:
Switched to a different node kind. This necessitated adding node stack functionality to treat all parse nodes that produce names uniformly, like we do for expressions, so there's a fair bit more churn in this patch now. There's probably some refactoring we ought to be doing in the node stack, but I think it can wait until we have a third class of node kinds we want to treat uniformly.